python/pyspark/sql/functions.py

```diff
@@ -1437,10 +1453,6 @@ def hash(*cols):
     'ascii': 'Computes the numeric value of the first character of the string column.',
     'base64': 'Computes the BASE64 encoding of a binary column and returns it as a string column.',
     'unbase64': 'Decodes a BASE64 encoded string column and returns it as a binary column.',
-    'initcap': 'Returns a new string column by converting the first letter of each word to ' +
-               'uppercase. Words are delimited by whitespace.',
-    'lower': 'Converts a string column to lower case.',
-    'upper': 'Converts a string column to upper case.',
```

Comment:
This had to stay in string functions!

Reply:
`initcap` was also defined explicitly below, with better documentation, so I preserved that instance. `lower` and `upper` have been introduced in the 1.3 API, and were being defined there as well, so I prioritised that instance because it used the correct `@since` number. Arguably we could've used a separate mapping like `string_functions_1_3` or something, but that didn't seem appropriate since the point of these is to minimise code.

Reply:
I was saying `lower` and `upper`. Looks it has been overwritten by this, so if we keep here, it keeps previous definition. Did you check which PR and JIRAs added `lower` and `upper`?

Also, strictly this should have not been removed in this PR as it doesn't target to remove overwritten functions. As you said, we should avoid such function definition way later.

Reply:
Also, if it was an exception, it had to describe it specifically.

```
    # name functions take a column name as their argument
    'lit': _lit_doc,
```

This doesn't look making sense if you read the codes from scratch.

Reply:
The java/scala API documentation says it was added in 1.3, but I just tracked down the JIRA/PR and it seems it actually was 1.0.

https://issues.apache.org/jira/browse/SPARK-1995
https://github.com/apache/spark/pull/936

As for removing overwritten functions, maybe it would've been better to make a separate PR, but the first fix did require removing them. When I changed the approach it seemed reasonable to keep the change, since the problem was obvious and easy to fix.